File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        int ans[] = new int[2];
        for (int i=0; i<nums.length; i++){
            int diff = target - nums[i];
            if(map.containsKey(diff)){
                int j = map.get(diff);
                ans[0] = (i<j)?i+1:j+1;
                ans[1] = (i>j)?i+1:j+1;
                return ans;
            }
            map.put(nums[i], i);
        }
        return ans;
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-0.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        
        ans = set()
        for i in range(len(nums)):
            target = -nums[i]
            hashset = set()

            for j in range(i+1, len(nums)):
                diff = target - nums[j]
                if diff in hashset:
                    tuplet = tuple(sorted([diff, nums[j], nums[i]]))
                    ans.add(tuplet)
                hashset.add(nums[j])
            
        return [list(tuplet) for tuplet in ans]